portfolio/src/main/webapp/script.js

```diff
@@ -38,18 +37,100 @@ function showModal(img) {
   modal.style.display = "block";
   modalImg.src = img.src;
   modalCaption.innerHTML = img.alt;
+  getServerData();
 }
 
 function closeModal() {
   const modal = document.getElementById("my-modal");
   modal.style.display = "none";
+  const maxElement = document.getElementById("quantity");
+  maxElement.value = "";
 }
 
 // Request content from server
 function getServerData() {
   console.log("Fetching data.");
-  const promise = fetch("/data").then(response => response.json()).then(greeting => {
-    console.log(greeting);
-    document.getElementById('data-container').innerText = greeting;
+  const maxElement = document.getElementById("quantity");
+  const maxToDisplay = maxElement.value;
+  const sort = document.getElementById("sort").value;
+  
+  const imgsrc = document.getElementById("modal-img").src;
+  const splitimg = imgsrc.split("/");
+  console.log(splitimg);
+  // Get image file name in img tag's src field (ex: flowers.jpg)
+  const splitname = splitimg[splitimg.length - 1].split(".");
+  // Get string before file format string (remove .jpg)
+  const img = splitname[0];
+
+  console.log(sort);
+  console.log(maxToDisplay);
+  
+  fetch("/comment?quantity=" + maxToDisplay + "&sort=" + sort + "&img=" + img).then(response => response.json()).then(jsonObj => {
+    const comments = jsonObj.comments;
+    console.log(comments[comments.length - 1]);
+    const total = jsonObj.total;
+    console.log(total);
+
+    const totalElement = document.getElementById("total");
+    totalElement.innerText = ("Comments (" + total + ")");
+
+    // Set maximum number of comments to show as the total number of comments (returned from fetch)
+    maxElement.max = total.toString();
+    maxElement.placeholder = total.toString();
+
+    // Add individual comments to html page
+    const list = document.getElementById("all-comments");
+    list.innerHTML = "";
+    comments.forEach((comment) => {
```

Comment:
We can also display timestamp (since we are already logging it)!

Reply:
The timestamps are just recorded as big numbers like: 1591316788695
I think to get a nicely formatted date, I need to log it differently but I can try to find a way!